### PR TITLE
Fix multigpu problem on v0.3.0

### PR DESCRIPTION
**Pull request by kamo-naoyuki**
Fix a bug when using Multi GPU for pytorch due to #484

### CODE REVIEW

**Comment by kamo-naoyuki** [comment]
I fixed only for PyTorch. The multi-GPU on chainer version still has problems, but I won't fix it. This is a little troublesome for me because I won't use chainer anymore. Anyone fix, please.

File: espnet/nets/pytorch_backend/e2e_asr.py [hunk 1]
```diff
@@ -251,6 +251,20 @@ def forward(self, xs_pad, ilens, ys_pad):
         else:
             logging.warning('loss (=%f) is not correct', loss_data)
 
+        # Note(kamo): In order to work with DataParallel, on pytorch==0.4,
+        # the return value must be torch.CudaTensor, or tuple/list/dict of it.
+        # Neither CPUTensor nor float/int value can be used
+        # because NCCL communicates between GPU devices.
+        if self.loss.is_cuda:
```
[the remark this answers is not in the document]
Comment (kamo-naoyuki) [reply espnet/nets/pytorch_backend/e2e_asr.py]: How is this?

[the remark this answers is not in the document]
Comment (kamo-naoyuki) [reply espnet/nets/pytorch_backend/e2e_asr.py]: https://github.com/espnet/espnet/blob/b2e6052037c0a8961caa8a244b7fd0823f056188/espnet/nets/pytorch_backend/e2e_asr.py#L258-L261

**Comment by sw005320** [comment]
> I fixed only for PyTorch. The multi-GPU on chainer version still has problems, but I won't fix it. This is a little troublesome for me because I won't use chainer anymore. Anyone fix, please.

@Fhrozen, can you do it?

**Comment by Fhrozen** [comment]
I am checking it. I will pull a request once completed and tested.

**Comment by kan-bayashi** [comment]
I will merge this PR.

**Pull request merged**